okio-zipfilesystem/src/jvmMain/kotlin/okio/zipfilesystem/ZipEntry.kt

```diff
@@ -55,46 +65,48 @@ internal class ZipEntry(
    *
    * @throws IllegalArgumentException if `value` is < 0 or > 0xFFFFFFFFL.
    */
-  val crc: Long,
+  val crc: Long = -1L,
 
   /**
    * Gets the compressed size of this `ZipEntry`.
    *
    * @return the compressed size, or -1 if the compressed size has not been
    * set.
    */
-  val compressedSize: Long,
+  val compressedSize: Long = -1L,
 
   /**
    * Gets the uncompressed size of this `ZipEntry`.
    *
    * @return the uncompressed size, or `-1` if the size has not been
    * set.
    */
-  val size: Long,
+  val size: Long = -1L,
 
   /**
    * Gets the compression method for this `ZipEntry`.
    *
    * @return the compression method, either `DEFLATED`, `STORED`
    * or -1 if the compression method has not been set.
    */
-  val compressionMethod: Int,
+  val compressionMethod: Int = -1,
 
-  val time: Int,
+  val time: Int = -1,
 
-  val modDate: Int,
+  val modDate: Int = -1,
 
   /**
    * Gets the extra information for this `ZipEntry`.
    *
    * @return a byte array containing the extra information, or `null` if
    * there is none.
    */
-  val extra: ByteString,
+  val extra: ByteString = ByteString.EMPTY,
```

Comment:
null again...

Reply:
I’m (foolishly?) preserving as much as possible of the original structure of this code in this PR, in hopes of getting all the tests in place before I do any real cleanup. The code is also pretty clumsy!